apps/api/src/server/applications/application/file-folders/folders.service.js

```diff
@@ -113,6 +112,23 @@ export const getDocumentsInFolder = async (folderId, pageNumber = 1, pageSize =
 	};
 };
 
+/**
+ * Returns a list of folderIds and their parentFolderIds
+ *
+ * @param {number} folderId
+ * @param {Array<{ id: number, parentFolderId: number | null }>} folderList
+ * @returns {Promise<Array<{ id: number, parentFolderId: number | null }>>}
+ */
+export const getChildFolders = async (folderId, folderList = []) => {
+	const currentLevelFolderList = await folderRepository.getFoldersByParentId(folderId, {
+		select: { id: true, parentFolderId: true }
+	});
+	folderList.push(...currentLevelFolderList);
+
+	await Promise.all(currentLevelFolderList.map((folder) => getChildFolders(folder.id, folderList)));
```

Comment:
Just thinking, `getChildFolders` returns a promise, so here you're creating an array of promises and then waiting for them all to resolve. But if say one of them fails, I think it will fail silently (it just won't push to `folderList`) which could be difficult to debug.

To get around it you could change `getChildFolders` so that rather than mutating its `folderList` argument, it just returns a new, updated list. Then you could use [Promise.allSettled](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/allSettled) here instead of `Promise.all` to keep track of the list of promises. If any fail you'll get a result in the `rejected` state and you can throw an error or deal with the error some other way.

This should work as it is though, so this could be an improvement for later. Maybe worth leaving a tech debt note. Let me know what you think though.

Reply:
nice one! Addressed that in latest commit